SDWebImage/SDImageCache.m

```diff
@@ -136,12 +136,28 @@ - (nullable NSString *)cachedFileNameForKey:(nullable NSString *)key {
     CC_MD5(str, (CC_LONG)strlen(str), r);
     NSURL *keyURL = [NSURL URLWithString:key];
     NSString *ext = keyURL ? keyURL.pathExtension : key.pathExtension;
-    NSString *filename = [NSString stringWithFormat:@"%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%@",
+    
+    NSString *filename = [NSString stringWithFormat:@"%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x",
                           r[0], r[1], r[2], r[3], r[4], r[5], r[6], r[7], r[8], r[9], r[10],
-                          r[11], r[12], r[13], r[14], r[15], ext.length == 0 ? @"" : [NSString stringWithFormat:@".%@", ext]];
+                          r[11], r[12], r[13], r[14], r[15]];    
+    if (ext.length > 222) {
+        unsigned char e[CC_MD5_DIGEST_LENGTH];
+        const char *strExtension = key.pathExtension.UTF8String;
```

Comment:
This `key.pathExtension` (including one in the bottom), should use `ext` local variable instead. Because we check to use NSURL or NSString Path for compatible reason. Don't use `key.pathExtension` directly.

Reply:
yeah thx mention @dreampiggy